mitodl/ocw-hugo-themes
# feat: lists of course resources grouped by learning resource types

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/751

#### What's this PR do?
- [x] List course resources at `base-url/resources/`
- [x] `learning_resource_types` taxonomy used, introduced in `ocw-hugo-projects` in this PR https://github.com/mitodl/ocw-hugo-projects/pull/197
- [x] Resources being shown are grouped by `Learning resource type`
- [x] Within a group, ordered alphabetically
- [x] Limited number of resources shown. for now limit is 10.
- [x] If more than the limit, "See all" link is shown that links to a page that lists all the resources of that type. 
- [x] Separate link (button) do download the resource.
- [x] Clicking on resource takes us to the resource page.

#### How should this be manually tested?
- If this PR https://github.com/mitodl/ocw-hugo-projects/pull/197 is merged then take a pull of `ocw-hugo-projects` locally. If not merged then checkout to its branch.
- Checkout to the branch of this PR.
- Build any course locally.
- Go to `http://localhost:3000/resources/`
- Verify that all the resources are being rendered properly.
- Verify that resources are shown according to `learning resource type`
- Verify that resources in each group are sorted alphabetically.
- Verify that upon clicking of each resource, we are redirected to its page.
- Verify that all the resources are downloadable.
- Verify that the correct thumbnail is shown.
- Verify that only/upto 10 resources are being shown at max.
- Verify that for more resources than 10, a `See All` link is being shown.
- Verify that we are able to click that link and view all the resources of that particular type.
- Any other use-case that needs to be tesed. 

#### Screenshots

URL: `http://*base-url*/resources/`
<img width="695" alt="image" src="https://user-images.githubusercontent.com/93309234/175292172-79dabf10-979b-4fb2-97e9-81633204b478.png">

<img width="703" alt="image" src="https://user-images.githubusercontent.com/93309234/175292221-49bcdf35-ccf9-4cf1-9392-34eed3f7fd5b.png">


URL: `http://*base-url*/learning_resource_types/lecture-videos/`
<img width="689" alt="image" src="https://user-images.githubusercontent.com/93309234/175292321-c698a54a-22a4-4838-940a-49d895305f56.png">

URL: `http://*base-url*/learning_resource_types/`
<img width="687" alt="image" src="https://user-images.githubusercontent.com/93309234/175292347-dc3acdd2-ae27-4032-8113-cbe7891fc663.png">
